### PR TITLE
Setup Continuous Integration & Quality Assurance pipeline

### PR DESCRIPTION
**Pull request by Angelus123**
**What this PR does** 
This PR is for setting up a testing environment and pipeline.
- [x] Configured Testing environment with Mocha, Chai 
- [x] GitHub Actions is configured as a continuous integration service used to build and test the app.
- [x] GitHub Actions  badge has been added to your repo

**How could you test?**
git clone
switch to branch ch-setup-ci-and-qa-pipeline
npm install
npm test

### CODE REVIEW

**Review by Kalimbak — APPROVED** [review]

File: src/documentation/index.js [hunk 1]
```diff
@@ -1,4 +1,4 @@
-import welcome from './welcome';
+import welcome from './welcome.js';
```
Comment (tjeaneric) [inline src/documentation/index.js]: You can remove this .js  if babel is working now.

**Review by kwizeradev — APPROVED** [review]

File: .github/workflows/phantom.yml [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+          node-version: 16
+          registry-url: https://npm.pkg.github.com/
+      - run: npm ci
+      - run: npm publish
+        env:
+<<<<<<< HEAD
+          NODE_AUTH_TOKEN: ${{secrets.GITHUB_TOKEN}}
+=======
+          NODE_AUTH_TOKEN: ${{secrets.GITHUB_TOKEN}}
+>>>>>>> 618b10d7f642f4af3ceaf40d5ba05f61b0db8394
```
Comment (furebo) [inline .github/workflows/phantom.yml]: Please resolve the conflicts.

**Review by furebo — CHANGES_REQUESTED** [review]
Hi, Please resolve the conflicts with the develop, and also squash the commits

File: .github/workflows/phantom.yml (hunk 2 above)
Comment (Angelus123) [reply .github/workflows/phantom.yml]: The conflict was resolved

**Review by Kalimbak — APPROVED** [review]

**Review by tjeaneric — APPROVED** [review]

File: 1 [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+# Next commands to do (19 remaining commands):
+#    squash ccdd21a Setting up Node.js REST API empty project
+#    squash 327fd38 dotenv
+# You are currently rebasing branch 'ch-setup-ci-and-qa-pipeline' on 'd955acc'.
+#
+# Changes to be committed:
+#	new file:   __test__/appTest.js
+#	modified:   src/app.js
+#	new file:   test/app.test.js
+#
```
Comment (furebo) [inline 1]: please remove this file !

**Review by furebo — CHANGES_REQUESTED** [review]

File: src/documentation/index.js (hunk 1 above)
Comment (Angelus123) [reply src/documentation/index.js]: I have already removed it

File: 1 (hunk 3 above)
Comment (Angelus123) [reply 1]: I did, Sir

**Review by furebo — APPROVED** [review]

**Review by furebo — APPROVED** [review]

**Pull request merged**